mantidimaging/core/operations/rotate_stack/rotate_stack.py

```diff
@@ -96,15 +95,94 @@ def _update_angle(angle, dropdown):
             angle.setEnabled(False)
 
 
-def _rotate_image_inplace(data, angle=None):
-    data[:, :] = rotate(data[:, :], angle)
+def _get_cardinal_angle(value) -> int:
+    """
+    Calculates the angle based on the input value, by adding 45 to the value modulo 360,
+    dividing it by 90, multiplying by 90, and then subtracting 90 if the value modulo 360 modulo
+    90 is equal to 45, otherwise subtracting 0 to get the cardinal angle the aspect ratio should be
+    rotated by.
+
+    param: value: angle of rotation
+    return: angle: cardinal angle of rotation
+    """
+    angle = ((value % 360 + 45) // 90 * 90 - (90 if (value % 360) % 90 == 45 else 0))
```

Comment:
If we have already normalised the range to 0--360 in `filter_func` would `round(x/90)*90` do the same?

Reply:
I think i'm going to stay with the existing solution such that cardinal angle rounds down if exactly at the half way point like discussed originally so if value is 45deg, 0 is returned, but if 45.001, 90 is returned. If using `round(<45> / 90) * 90`, I would get 90 instead of 0.

Reply:
For me `round(45/90)` is giving `0`. The docs say it uses round-half-even, so it should prefer 0 and 180 when on the boundary. That means avoiding the aspect ratio change unless we are closer to 90 or 270 which seems reasonable.

Reply:
Using `round(value/90) * 90` fails for cases 135 and 315 for me, setting aspect ratios to the next aspect ratio, 180 and 0 respectively where value is set to 135 will change aspect ratio to 180 instead of leaving at 90 and 315 to 0 instead of leaving at 270.